net/graphsync_fetcher_test.go

```diff
@@ -369,32 +252,19 @@ func TestGraphsyncFetcher(t *testing.T) {
 	})
 
 	t.Run("value returned with non block format", func(t *testing.T) {
-		notABlock := types.NewMsgs(1)[0]
-		notABlockObj, err := notABlock.ToNode()
-		require.NoError(t, err)
-
-		notABlockCid, err := notABlock.Cid()
-		require.NoError(t, err)
-		originalCids := types.NewTipSetKey(notABlockCid)
-
-		stubs := []requestResponse{
-			{
-				fakeRequest{pid0, cidlink.Link{Cid: notABlockCid}, layer1Selector},
-				fakeResponse{blks: []format.Node{notABlockObj}},
-			},
-		}
-		mgs := &mockableGraphsync{stubs: stubs, t: t, store: bs}
-
+		gen := builder.NewGenesis()
+		mgs := newMockableGraphsync(ctx, builder, bs, t)
+		mgs.stubResponseWithUndecodableBlocks(pid0, gen.At(0).Cid(), layer1Selector, []cid.Cid{gen.At(0).Cid()})
```

Comment:
In this case I think it would be slightly better to be able to specify the response node explicitly. That would make it flexible for other responses too, like a block that does decode but fails syntactic or semantic validation.